bee-metrics/src/metrics.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Module containing some predefined useful metrics.
+
+pub use prometheus_client::metrics::{counter, gauge};
+use prometheus_client::{
+    encoding::text::{EncodeMetric, Encoder},
+    metrics::{gauge::Gauge, MetricType},
+};
+use tokio::process::Command;
+
+/// Type used to track and update metrics about an specific OS process.
+#[derive(Clone)]
+pub struct ProcessMetrics {
+    mem: Gauge,
+    cpu: Gauge<f64>,
+    pid: String,
+}
+
+impl ProcessMetrics {
+    /// Creates a new set of metrics for the desired PID.
+    pub fn new(pid: u32) -> Self {
+        Self {
+            mem: Gauge::default(),
+            cpu: Gauge::default(),
+            pid: pid.to_string(),
+        }
+    }
+
+    /// Obtains the metrics tracked by this value.
+    pub fn metrics(&self) -> (MemoryUsage, CpuUsage) {
+        (MemoryUsage(self.mem.clone()), CpuUsage(self.cpu.clone()))
+    }
+
+    /// Updates the metrics values.
+    ///
+    /// The value is not updated if the command used to retrieve the new value:
+    ///  - cannot be spawned,
+    ///  - returns an unsuccessful exit code,
+    ///  - its output cannot be parsed.
+    pub async fn update(&self) {
```

Comment:
Would it make sense to return an `Result<(), Error>` here? I feel like a lot of different things can go wrong that we might want to log.

Reply:
Yeah that might be the case but I wonder what kind of error would this be. Because it has to be able to represent:
- `std::io::Error` if the process fails.
- `std::string::FromUtf8Error` if the output is not utf8.
- Some custom error if we cannot parse the output.
And I don't know if I'd log that, or at least not periodically.

Reply:
If we are not going to log it, I think it’s fine how it is right now.